### PR TITLE
[vcr-2.0] Impl FIND

### PR DESCRIPTION
**Pull request by snalli**
Implements findMissingKeys and findKey for backupV2

### CODE REVIEW

File: .github/workflows/github-actions.yml [hunk 1]
```diff
@@ -52,6 +52,11 @@ jobs:
       - name: Upload coverage to Codecov
         uses: codecov/codecov-action@v1
 
+      - name: Install and run Azurite
```
[the remark this answers is not in the document]
Comment (snalli) [reply .github/workflows/github-actions.yml:55]: I'll remove the installation of Azurite and the @before method of unit-tests will prevent V2 tests from running.

File: ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureCloudDestinationSync.java [hunk 2]
```diff
@@ -489,6 +490,23 @@ protected boolean isDeleted(Map<String, String> cloudMetadata) {
         && Long.parseLong(cloudMetadata.get(CloudBlobMetadata.FIELD_DELETION_TIME)) != Utils.Infinite_Time;
   }
 
+  /**
+   * For testing, ironic that its used for testing and not for prod
+   * @param blobId
+   * @return
+   */
+  @Override
+  public boolean doesBlobExist(BlobId blobId) {
```
[the remark this answers is not in the document]
Comment (snalli) [reply ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureCloudDestinationSync.java:499]: Going from using Microsoft's well-written storage emulator to an archaic mock layer is a huge step backward. 
V1 tests can continue using the mock layer.
For V2, I'll continue using the emulator. It is immensely useful in catching issues that the mock layer never can. 
For eg. did you know container names can only be between 3-63 char and include only lower case alphabets ?

**Review by Arun-LinkedIn — APPROVED** [review]

**Review by JingQianCloud — APPROVED** [review]

File: .github/workflows/github-actions.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (JingQianCloud) [reply .github/workflows/github-actions.yml:55]: Didn't realize it's the emulator. That's fine.

**Comment by snalli** [comment]
Merging as all comments addressed and no more pending comments.

**Pull request merged**